docs/devel/testing.md

```diff
@@ -0,0 +1,66 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/devel/testing.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+Kubernetes Commit Queue Testing
+===============================
+
+A quick overview of how we add, remove and recycle tests from CI.
+
+## What is CI?
+
+Throughout this document we will refer to CI as any suite of e2e tests that can potentially hold up the submit queue, this means Kubernetes PRs must pass these tests prior to getting merged.
+
+## Adding a test to CI
+
+When first adding a test it should *not* go straight into CI, because failures block ordinary development. A test should only be added to CI after is has been running in some non-CI suite long enought to establish a track record showing that the test does not fail when run against *working* software. A suite named `flaky` exists, and can be overloaded to mean `experimental` and used for this reason (can it really?). In addition to this track record, consider the following as requirements:
```

Comment:
> Does it make sense to version tests in parallel to APIs, where alpha tests are quarantined and don't break CI, but beta & core (?) APIs do? (This really is a question; I don't know enough to say, it just occurred to me.)

I think core should be ci. everything else should be some TBD solution that gives us results without throwing the team into panic. 

It isn't clear why a failure in something like a volume plugin test (1 of like 5 volume plugins out there) should hold up changes to mesos. something like the scheduler breaking is important, but most developers don't care about other sprinkles we put on top of kubernetes. Heapster, GCE L7, rkt all those shouldn't halt developer progress IMO. 

On afore mentioned TDB solution would be to run out of band suites, and hook up email notifications so the parties that care get notified. I view this as a seperate problem from: what do we run in CI?